### PR TITLE
DIV-4619: changed html code to use <fieldset/> and <legend/> in form

### PR DESCRIPTION
**Pull request by yarpo**
# Description

Accessibility improvement. Updated HTML code by adding <legend/> to <fieldset/> tags so it can be read by screen readers for people less able to see.

Fixes # (issue)
https://tools.hmcts.net/jira/browse/DIV-4619

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

manually, using mac screen reader

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] ~I have commented my code, particularly in hard-to-understand areas~
- [ ] ~I have made corresponding changes to the documentation~
- [x] My changes generate no new warnings
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: steps/claim-costs/ClaimCosts.html [hunk 1]
```diff
@@ -40,38 +40,59 @@
...
-      ]
-     ) }}
+  <fieldset>
+    <legend class="visuallyhidden">{{ content.title | safe}}</legend>
+
+    <div class="form-group {{ errorClass(fields.dnCosts.claimCosts) }}">
+      {{ errorsFor(fields.dnCosts.claimCosts) }}
+      {{
+        selectionButton(
+          field = fields.dnCosts.claimCosts,
```
Comment (hosainnet) [inline steps/claim-costs/ClaimCosts.html]: Seems like we missed the content.fields.dnCosts.title param here

File: steps/claim-costs/ClaimCosts.html [hunk 2]
```diff
@@ -40,38 +40,59 @@
...
-          id="{{ fields.dnCosts.costsDifferentDetails.id }}"
-          name="{{ fields.dnCosts.costsDifferentDetails.id }}"
-          rows="4">{% if fields.dnCosts.costsDifferentDetails.value %}{{ fields.dnCosts.costsDifferentDetails.value }}{% endif %}</textarea>
-      </div>
-    {% endif %}
+    <div class="form-group {{ errorClass(fields.dnCosts.claimCosts) }}">
+      {{ errorsFor(fields.dnCosts.claimCosts) }}
+      {{
+        selectionButton(
+          field = fields.dnCosts.claimCosts,
```
Comment (hosainnet) [inline steps/claim-costs/ClaimCosts.html]: And here

File: steps/claim-costs/ClaimCosts.html [hunk 3]
```diff
@@ -40,38 +40,59 @@
...
+        name="{{ fields.dnCosts.costsDifferentDetails.id }}"
+        rows="4">{% if fields.dnCosts.costsDifferentDetails.value %}{{ fields.dnCosts.costsDifferentDetails.value }}{% endif %}</textarea>
+    </div>
+    {% endif %}
 
+    <div class="form-group {{ errorClass(fields.dnCosts.claimCosts) }}">
+      {{ errorsFor(fields.dnCosts.claimCosts) }}
+      {{
+        selectionButton(
+          field = fields.dnCosts.claimCosts,
```
Comment (hosainnet) [inline steps/claim-costs/ClaimCosts.html]: And here

File: steps/claim-costs/ClaimCosts.html (hunk 1 above)
Comment (alexwakeman) [reply steps/claim-costs/ClaimCosts.html]: I think they map over to the fields object @hosainnet - given this has been tested already as well

Comment (hosainnet) [reply steps/claim-costs/ClaimCosts.html]: Ah sorry the signture selectionButton is different than selectionButtons. 
@yarpo maybe for the future we should consider fixing selectionButtons to be better with screenreaders if that's the issue? As it already adds a visually-hidden title etc https://github.com/hmcts/look-and-feel/blob/e858809aaefb366c174e63e9988dc83314ee5e88/templates/look-and-feel/components/fields.njk#L127

**Comment by hosainnet** [comment]
So what do we do with this given that #353 is closed?

**Comment by hosainnet** [comment]
Closing the PR for now - will be re-opened once govuk is merged (will keep the branch open)

**Pull request closed without merging**